From c7e800738a6b894faf91047ea8a01489dba1cc9d Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:22 -0400
Subject: [ethernet] i40e: count timeout events

Message-id: <1409658042-22689-22-git-send-email-sassmann@redhat.com>
Patchwork-id: 90218
O-Subject: [RHEL7.1 PATCH 021/281] i40e: count timeout events
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

The ethtool -S statistics should have a counter for
tx timeouts in order to better help inform the masses.

Change-ID: Ice4b20ed4a151509f366719ab105be49c9e7b2b4
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit a47a15f497cac2c94eb88f4eb11929564ea856d3)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index c6471bc..cb6fc95 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -87,6 +87,7 @@ static struct i40e_stats i40e_gstrings_stats[] = {
  I40E_PF_STAT("illegal_bytes", stats.illegal_bytes),
  I40E_PF_STAT("mac_local_faults", stats.mac_local_faults),
  I40E_PF_STAT("mac_remote_faults", stats.mac_remote_faults),
+ I40E_PF_STAT("tx_timeout", tx_timeout_count),
  I40E_PF_STAT("rx_length_errors", stats.rx_length_errors),
  I40E_PF_STAT("link_xon_rx", stats.link_xon_rx),
  I40E_PF_STAT("link_xoff_rx", stats.link_xoff_rx),
-- 
1.7.1